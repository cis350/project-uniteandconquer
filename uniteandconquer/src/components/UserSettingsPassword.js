/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import Modal from 'react-modal';
import SidebarSettings from './SidebarSettings';
import '../assets/UserSettingsPassword.css';

const UserDB = require('../modules/UserDB');

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    background: '#FFD9A0',
  },
};

function UserSettingsPassword() {
  const [password, setPassword] = useState('');
  const [oldPassword, setOldPassword] = useState('');
  const myStorage = window.sessionStorage;
  const [modalIsOpen, setIsOpen] = useState(false);
  const [note, setNote] = useState(null);
  const closeModal = () => {
    setIsOpen(false);
  };

  const showModal = () => {
    setIsOpen(true);
  };

  const updatePasswordWithVerify = async (userID) => {
    await UserDB.modifyUser(userID, 'password', password, oldPassword, (success, err) => {
      if (success) {
        setNote('Password updated successfully');
        showModal();
      } else {
        setNote('Password updated unsuccessfully');
        showModal();
      }
    });
  };

  const updatePassword = () => {
    const userID = myStorage.getItem('UserID');
    updatePasswordWithVerify(userID);
  };
  return (
    <div>
      <div>
        <div>
          <Modal
            isOpen={modalIsOpen}
            onRequestClose={closeModal}
            style={customStyles}
            contentLabel="Example Modal"
            ariaHideApp={false}
          >
            <div className="modalContent">
              <h2>{note}</h2>
              <button className="modalButton" type="button" onClick={closeModal}>Close</button>
            </div>
          </Modal>
        </div>
      </div>
      <div className="user-settings-password">
        <SidebarSettings />
        <div>
          <div className="menu-title"><h1>Password</h1></div>
          <div className="post-input">
            <div className="all-fields">
              {' '}
              <div className="post-fields">
                <div className="post-field">
                  <div className="label">Old Password</div>
                  <input onChange={(e) => setOldPassword(e.target.value)} />
                </div>
                <div className="post-field">
                  <div className="label">New Password</div>
                  <input onChange={(e) => setPassword(e.target.value)} />
                </div>
                <div className="post-field">
                  <div className="label">Confirm New Password</div>
                  <input onChange={(e) => setPassword(e.target.value)} />
                </div>
                <button className="update" type="button" onClick={updatePassword}>
                  Change Password
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default UserSettingsPassword;
