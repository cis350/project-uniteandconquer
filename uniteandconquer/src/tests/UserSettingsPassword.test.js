/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import UserSettingsPassword from '../components/UserSettingsPassword';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('user settings for password content (header)', () => {
  render(<UserSettingsPassword />);
  const headerElement = screen.getByText('Password');
  expect(headerElement).toBeInTheDocument();
});

test('change old password', () => {
  render(<UserSettingsPassword />);
  const oldPassword = screen.getByTestId('old-pw-input');
  fireEvent.change(oldPassword, { target: { value: 'asdf!234BD3' } });
  expect(oldPassword.value).toBe('asdf!234BD3');
  const newPassword = screen.getByTestId('new-pw-input');
  fireEvent.change(newPassword, { target: { value: '34321CKJF!kda' } });
  expect(newPassword.value).toBe('34321CKJF!kda');
  const confirmNewPassword = screen.getByTestId('confirm-new-pw-input');
  fireEvent.change(confirmNewPassword, { target: { value: '34321CKJF!kda' } });
  expect(confirmNewPassword.value).toBe('34321CKJF!kda');
  fireEvent.click(screen.getByText('Change Password'));
});

test('snapshot user settings for password', () => {
  const component = renderer.create(<UserSettingsPassword />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
