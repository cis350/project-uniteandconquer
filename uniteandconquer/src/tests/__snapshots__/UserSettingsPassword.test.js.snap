// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot user settings for password 1`] = `
<div
  className="user-settings-password"
>
  <div
    className="sidebarSettings"
  >
    <div
      className="greeting"
    >
      <i
        className="far fa-user-circle fa-2x"
      />
       
      Hi, Jeremy
    </div>
    <div
      className="main-page"
      data-testid="main-page-header"
    />
    <div
      className="personal-information"
      data-testid="personal-information-header"
    />
    <div
      className="password"
      data-testid="password-header"
    />
  </div>
  <div>
    <div
      className="menu-title"
    >
      <h1>
        Password
      </h1>
    </div>
    <div
      className="post-input"
    >
      <div
        className="all-fields"
      >
         
        <div
          className="post-fields"
        >
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              Old Password
            </div>
            <input
              data-testid="old-pw-input"
              onChange={[Function]}
            />
          </div>
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              New Password
            </div>
            <input
              data-testid="new-pw-input"
              onChange={[Function]}
            />
          </div>
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              Confirm New Password
            </div>
            <input
              data-testid="confirm-new-pw-input"
              onChange={[Function]}
            />
          </div>
          <button
            className="update"
            onClick={[Function]}
            type="button"
          >
            Change Password
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
